src/languages/typescript.js

```diff
@@ -36,17 +36,15 @@ export default function(hljs) {
   };
 
   const INTERFACE = {
-    begin: [
+    match: [
       /interface/,
       /\s+/,
       IDENT_GEN
     ],
-    beginScope: {
+    scope: {
       1: "keyword",
       3: "title"
     },
-    end: /\{|\s/,
-    excludeEnd: true,
     keywords: {
       keyword: 'interface extends',
```

Comment:
Oh, n/m this isn't going to get the "extends" though... we'd need a second match variant to handle `interface xyz extends abc`  That is the current preference over the more ambiguous "look for a {" which can have false positives in other languages.  Again you should have examples to refer to in `ruby` or `wren`... you just need `variants` to house two multi-matchers... 

And of course we need some markup tests to catch things like this for us. :)